### PR TITLE
Docs: cassandra_KVS_config should have more information on ssl and leader block

### LINKED ISSUE #783: Cassandra KVS configuration docs improvements
There are a number of things to make clearer on the [Cassandra Key Value Service Configuration](https://palantir.github.io/atlasdb/html/configuration/cassandra_KVS_configuration.html) page. Listing some here:

- the `ssl` property is still described in the bottom section even though it's no longer listed as an example. We should probably keep `ssl` as an example and have a comment saying it's deprecated.
- add docs about what a leader block is, and it's also weird that the only location where we describe the leader block config is on a cassandra-specific configuration page
- Should change the default example of `leaders` to be `https://host[123]:3828`. Having SSL enabled and then specifying leaders with "http" will break when AtlasDB clients try talking to each other.
 - We should probably file a bug about improving that error message since it gives a cryptic “EOF exception use \n" message.

### PR DESCRIPTION
**Pull request by hsaraogi**
Fixes #783 
The leader block information is added here. We can move that to a separate section and link CassandraKVSConfig and PostgresKVSConfig pages to it.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/787)

### CODE REVIEW

File: docs/source/configuration/cassandra_KVS_configuration.rst [hunk 1]
```diff
@@ -51,22 +51,35 @@ in the leaders list. If you do not specify a lock creator, the leaders block sho
...
+          - https://host2:3828
+          - https://host3:3828
 
 .. _cass-config-ssl:
 
 Communicating Over SSL
 ======================
 
-Atlas currently supports two different ways of specifying SSL options in the Cassandra KVS configuration: The ``sslConfiguration`` block and the deprecated ``ssl`` property.  Both means are supported but ``sslConfiguration`` is preferred and will always be respected in favor of ``ssl`` when both are specified.  In the future, support for ``ssl`` will be removed.
+Atlas currently supports two different ways of specifying SSL options in the Cassandra KVS configuration: The ``sslConfiguration`` block and the ``ssl`` property.  Both means are supported but ``sslConfiguration`` is preferred and will always be respected in favor of ``ssl`` when both are specified.  In the future, support for ``ssl`` will be removed.
```
Comment (joelea) [inline docs/source/configuration/cassandra_KVS_configuration.rst]: I think this is still wrong.

File: docs/source/configuration/cassandra_KVS_configuration.rst [hunk 2]
```diff
@@ -77,3 +90,23 @@ ssl
 ---
 
 This property is a boolean value saying whether or not to use ssl.  When ``true``, it will use java system properties that are passed in as jvm arguments to determine how to set up the ssl connection.  For example, you would use the jvm option ``-Djavax.net.ssl.trustStore=<path-to-truststore>`` to tell atlas where to find the truststore to use.
+
+Leader
+======
+
+The leader block is where you specify the leaders(all hosts that participate in leader election) for paxos and a lockCreator for creating the locks table.
```
Comment (joelea) [inline docs/source/configuration/cassandra_KVS_configuration.rst]: should be "leaders ("

Comment (joelea) [inline docs/source/configuration/cassandra_KVS_configuration.rst]: "schema mutation lock table"

File: docs/source/configuration/cassandra_KVS_configuration.rst [hunk 3]
```diff
@@ -77,3 +90,23 @@ ssl
...
+
+Required parameters:
+
+- ``quorumSize`` : Number of hosts that form majority. This number must be greater than half of the total number of hosts.
```
Comment (joelea) [inline docs/source/configuration/cassandra_KVS_configuration.rst]: "a majority"

**Comment by joelea** [comment]
Looks good, like the table.

Can you add release notes for this as well?

File: docs/source/configuration/cassandra_KVS_configuration.rst (hunk 1 above)
Comment (hsaraogi) [inline docs/source/configuration/cassandra_KVS_configuration.rst]: Is "In the future, support for ``ssl`` will be removed." wrong?

**Comment by rhero** [comment]
Can you add the PR link to the release notes? Looks good otherwise.

**Comment by tpetracca** [comment]
Should we maybe make the leader info it's own page?  It's not technically CassandraKVS specific.

**Comment by hsaraogi** [comment]
@tpetracca : I wanted to do that, but then we will also have to link the C*KVS_config and postgres_config pages to the Leader block config page, right.

**Pull request merged**